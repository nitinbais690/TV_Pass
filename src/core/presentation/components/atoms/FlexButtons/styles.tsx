import { StyleSheet } from 'react-native';
import { appPaddingValues } from 'core/styles/AppStyles';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
    spanEven: {
        flex: 1,
    },
    marginEnd: {
        marginEnd: appPaddingValues.xxxs,
    },
});
