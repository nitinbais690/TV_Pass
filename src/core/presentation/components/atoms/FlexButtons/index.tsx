import React from 'react';
import { View } from 'react-native';
import PrimaryButton from '../PrimaryButton';
import SecondaryButton from '../SecondaryButton';

import { styles } from './styles';

export default function FlexButtons(props: FlexButtonsProps) {
    return (
        <View style={[styles.container, props.containerStyle]}>
            <SecondaryButton
                containerStyle={[styles.marginEnd, styles.spanEven]}
                title={props.secondaryButtonText}
                onPress={props.onPressSecondary}
                secondaryTestID={props.secondaryTestID}
                secondaryAccessibilityLabel={props.secondaryAccessibilityLabel}
            />
            <View style={[styles.spanEven]}>
                <PrimaryButton
                    primaryButtonIcon={props.primaryButtonIcon}
                    containerStyle={{ zIndex: 0 }}
                    title={props.primryButtonText}
                    onPress={props.onPressPrimary}
                    primaryTestID={props.primaryTestID}
                    primaryAccessibilityLabel={props.primaryAccessibilityLabel}
                />
                {props.primaryButtonOverlay}
            </View>
        </View>
    );
}

export interface FlexButtonsProps {
    onPressPrimary: () => void;
    onPressSecondary: () => void;
    primryButtonText: string;
    secondaryButtonText: string;
    primaryButtonIcon?: any;
    primaryButtonOverlay?: React.ReactNode;
    containerStyle?: {};
    primaryTestID?: string;
    primaryAccessibilityLabel?: string;
    secondaryTestID?: string;
    secondaryAccessibilityLabel?: string;
}
