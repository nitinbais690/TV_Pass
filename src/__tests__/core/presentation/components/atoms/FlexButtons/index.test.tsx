import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import FlexButtons from 'core/presentation/components/atoms/FlexButtons';
import { StyleSheet } from 'react-native';

jest.mock('core/presentation/hooks/use-app-colors');

describe('flexButtons', () => {
    const primaryText: string = 'Watch';
    const secondaryText: string = 'Play Trailer';
    const primaryCallback = jest.fn();
    const secondaryCallback = jest.fn();

    test('renders correctly', () => {
        const { container } = render(
            <FlexButtons
                primryButtonText={primaryText}
                secondaryButtonText={secondaryText}
                onPressPrimary={primaryCallback}
                onPressSecondary={secondaryCallback}
            />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with custom style', () => {
        const styles = StyleSheet.create({
            container: {
                padding: 16,
            },
        });

        const { container } = render(
            <FlexButtons
                primryButtonText={primaryText}
                secondaryButtonText={secondaryText}
                onPressPrimary={primaryCallback}
                onPressSecondary={secondaryCallback}
                containerStyle={styles.container}
            />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with primary icon', () => {
        const { container } = render(
            <FlexButtons
                primryButtonText={primaryText}
                secondaryButtonText={secondaryText}
                onPressPrimary={primaryCallback}
                onPressSecondary={secondaryCallback}
                primaryButtonIcon={'Icon'}
            />,
        );
        expect(toJSON(container)).toMatchSnapshot();
    });
});
