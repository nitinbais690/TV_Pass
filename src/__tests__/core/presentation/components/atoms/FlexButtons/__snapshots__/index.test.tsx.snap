// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flexButtons render with custom style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Object {
          "padding": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
          "marginEnd": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        hasTVPreferredFocus={true}
        isTVSelectable={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#1A1D1E",
              "borderColor": "#35393F",
              "borderRadius": 35,
              "borderStyle": "solid",
              "borderWidth": 1.457142857142857,
              "elevation": 6,
              "flexDirection": "row",
              "height": 70,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Regular",
                "fontSize": 18,
                "fontWeight": "600",
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Play Trailer
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "zIndex": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hasTVPreferredFocus={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4294929710,
                4290124297,
              ]
            }
            endPoint={
              Object {
                "x": 0.5,
                "y": 1,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0.5,
                "y": 0,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3B4046",
                "borderColor": "#2089dc",
                "borderRadius": 35,
                "borderWidth": 0,
                "elevation": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "ProximaNova-Regular",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Watch
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`flexButtons render with primary icon 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
          "marginEnd": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        hasTVPreferredFocus={true}
        isTVSelectable={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#1A1D1E",
              "borderColor": "#35393F",
              "borderRadius": 35,
              "borderStyle": "solid",
              "borderWidth": 1.457142857142857,
              "elevation": 6,
              "flexDirection": "row",
              "height": 70,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Regular",
                "fontSize": 18,
                "fontWeight": "600",
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Play Trailer
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "zIndex": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hasTVPreferredFocus={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4294929710,
                4290124297,
              ]
            }
            endPoint={
              Object {
                "x": 0.5,
                "y": 1,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0.5,
                "y": 0,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3B4046",
                "borderColor": "#2089dc",
                "borderRadius": 35,
                "borderWidth": 0,
                "elevation": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 8,
              }
            }
          >
            <Icon
              style={
                Object {
                  "marginEnd": 14.571428571428571,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "ProximaNova-Regular",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Watch
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`flexButtons renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "flex": 1,
          "marginEnd": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        hasTVPreferredFocus={true}
        isTVSelectable={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#1A1D1E",
              "borderColor": "#35393F",
              "borderRadius": 35,
              "borderStyle": "solid",
              "borderWidth": 1.457142857142857,
              "elevation": 6,
              "flexDirection": "row",
              "height": 70,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Regular",
                "fontSize": 18,
                "fontWeight": "600",
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Play Trailer
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "borderRadius": 3,
            "zIndex": 0,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          hasTVPreferredFocus={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <BVLinearGradient
            colors={
              Array [
                4294929710,
                4290124297,
              ]
            }
            endPoint={
              Object {
                "x": 0.5,
                "y": 1,
              }
            }
            locations={null}
            startPoint={
              Object {
                "x": 0.5,
                "y": 0,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3B4046",
                "borderColor": "#2089dc",
                "borderRadius": 35,
                "borderWidth": 0,
                "elevation": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingHorizontal": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "ProximaNova-Regular",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Watch
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;
